### ISSUE TITLE
Add A65 kernel StageNode library for common graph ops (NMS, TopK, filters, coord transforms)

### ISSUE THREAD

**Issue by Onat-Inak** [sima-neat/core#210]
## What

Ship a catalog of A65-optimized `StageNode`s inside core that customers can plug into a `Graph` without writing GStreamer plugins. First batch: a small set of ops that apps already re-implement by hand today — NMS, TopK, ScoreFilter, ClassFilter, CoordsUnletterbox. Each kernel lives under `graph::nodes::ops::*` with a `StageExecutor` + unit tests + a numeric benchmark baseline. Pair the catalog with one tutorial chapter showing them composed in a realistic detection pipeline.

## Why

Customers doing postprocessing outside the canonical `SimaBoxDecode` fast path keep re-implementing the same ops:
- Classic NMS for models that emit raw logits and skip `BoxDecode`.
- TopK over classifier outputs (no framework support today).
- Score and class filtering for custom detection thresholds.
- Letterbox coordinate recovery to map preproc-space outputs back to source-image pixels.

Current options are all poor:
- **Write a GStreamer plugin.** Real barrier — caps negotiation, plugin registration, buffer lifecycle. Most customers never will.
- **Implement in app code.** Duplicated work across `sima-neat/apps`, no shared test surface, no benchmark anchor.
- **Use `stages::*` helpers.** Scoped to model-phase execution (Preproc / Infer / MLA / BoxDecode), not user-defined ops.

The `graph::nodes::StageNode` + `StageExecutor` mechanism is purpose-built for pure A65 kernels inside a `Graph`, but nothing ships in core today — only framework-internal scaffolding like `StampFrameId`, `StreamScheduler`, `FanOut`, `JoinBundle`, `Map`. There's a gap between "we have the vehicle" and "customers can benefit from it."

## Proposed Approach

**Namespace**: `simaai::neat::graph::nodes::ops::*` — separate from the existing `stages::*` one-shot helpers to avoid confusion.

**First batch (3-5 ops), picked by real-world duplication count in `sima-neat/apps`**:
- `NMS` — class-aware greedy NMS over `(score, class_id, [x1,y1,x2,y2])` records.
- `TopK` — select top-K highest-score entries; useful for classification postproc and detector pre-NMS trimming.
- `ScoreFilter` — drop entries below a score threshold.
- `ClassFilter` — allowlist / blocklist class IDs.
- `CoordsUnletterbox` — undo preproc letterboxing to recover source-image coordinates.

**Per-kernel shape**:
- `StageExecutor` subclass + `StageNode` wrapper.
- Unit tests (`tests/graph/nodes/ops/`) covering standard paths + edge cases (empty input, K > N, identical scores).
- µs-per-N benchmark baseline in a per-kernel README; "optimized" anchored to a concrete number, not a claim. NEON where it pays for itself, plain C++ where it doesn't.

**Tutorial**: one new chapter under `tutorials/` composing the kernels end-to-end:
```text
[Preproc PipelineNode] → [Model PipelineNode] → [NMS StageNode] → [TopK StageNode] → [Output]
```
Chapters 012 (`build_a_custom_data_graph`) and 013 (`embed_model_inside_graph`) are prerequisites.

## Scope Boundary

**Graph-only.** These kernels do not work inside a `Session` — `Session` is GStreamer-backed, and a `StageNode` is pure A65 C++ with no GStreamer element. Users on Session-only pipelines either migrate to `Graph` or wrap a Session as a `PipelineNode` inside a Graph (chapter 013 pattern).

Session-side parity is **out of scope** for this ticket. If we later want these ops available inside a `Session`, that is a separate piece of work (GStreamer plugin packaging of each op) and should be filed separately.

## Acceptance Criteria

- [ ] Between 3 and 5 kernels land under `graph::nodes::ops::*`, each as a `StageExecutor` + `StageNode` wrapper.
- [ ] Unit tests for each kernel under `tests/graph/nodes/ops/`, covering empty input, large-N, edge cases.
- [ ] Each kernel has a documented µs-per-N benchmark baseline committed alongside its tests (N chosen per op).
- [ ] One new tutorial chapter composes the kernels in a realistic detection graph and runs end-to-end on Modalix.
- [ ] At least one example under `sima-neat/apps` is migrated to use the new kernels in place of a hand-rolled NMS or TopK — dogfood check that the API is actually usable externally.

## Dependencies

- Builds on the existing `graph::nodes::StageNode` / `StageExecutor` infrastructure (already in core).
- Tutorial chapter depends on tutorial chapter 012 (custom graph) and 013 (embed model in graph), which land in PR #177.

Internal tracking ticket [NEAT-35](https://sima-ai.atlassian.net/browse/NEAT-35)

**Comment by shanguangzai** [comment]
Jira linkage migrated from `AUTO-2109` to `NEAT-35`.
